### PR TITLE
chore(auto-update): do not show error toast when silent update fails

### PR DESCRIPTION
**Pull request by gribnoysup**
Super small thing I spotten only now when double-checking windows installation: sometimes autoupdater fails to download the update (for example there is a github network issue), in that case it makes sense to show the error for the manual update because we are showing the progress bar, but it seems to me that we should probably ignore it for the "silent" update. There is no indication for users that there is an update happening, so the error looked a bit out of place in that case.

We haven't really documented this, so open to feedback on this, happy to close if it doesn't make sense to change this

### CODE REVIEW

File: packages/compass/src/main/handle-uncaught-exception.ts [hunk 1]
```diff
@@ -1,8 +1,6 @@
 import { handleError } from './handle-error';
 
 async function handleUncaughtException(err: Error): Promise<void> {
-  // eslint-disable-next-line no-console
-  console.error('handling uncaughtException', err);
```
Comment (gribnoysup) [inline packages/compass/src/main/handle-uncaught-exception.ts:5]: Drive-by, `handleError` already prints the message through console.error with a better message

**Comment by paula-stacho** [comment]
Would the user be notified in some other way if say, the autoupdate is persistently failing for them, and they find themselves on a really old version, or are missing important security updates?

**Comment by addaleax** [comment]
Yeah, I had assumed that this was intentional in the design because of what @paula-stacho just described.

**Comment by gribnoysup** [comment]
This is a temporary issue most of the time caused by intermittent network issues or not all our backend update nodes being up to date on available releases (this is a known issue of the backend that we never planned to address AFAIU). If we want to change the old behavior and start notifying users about update errors, we probably should make the download logic more robust (and maybe the toast more actionable), I'll close this PR and open tickets for this
